File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/View/EventTest.java
package com.spvessel.spacevil.View;

import java.awt.Color;

import com.spvessel.spacevil.ActiveWindow;
import com.spvessel.spacevil.BlankItem;
import com.spvessel.spacevil.Frame;
import com.spvessel.spacevil.ItemsRefreshManager;
import com.spvessel.spacevil.TitleBar;
import com.spvessel.spacevil.Decorations.Border;
import com.spvessel.spacevil.Decorations.CornerRadius;
import com.spvessel.spacevil.Decorations.Style;
import com.spvessel.spacevil.Flags.InputEventType;
import com.spvessel.spacevil.Flags.KeyCode;
import com.spvessel.spacevil.Flags.MSAA;

public class EventTest extends ActiveWindow {

    private BlankItem getBlankItem(String name) {
        BlankItem blank = new BlankItem();
        blank.setItemName(name);
        blank.setStyle(Style.getFrameStyle());
        blank.setMargin(50, 50, 50, 50);
        blank.setBorder(new Border(new Color(0, 162, 232), new CornerRadius(10), 3));
        blank.setBackground(255, 255, 255, 50);
        blank.eventMouseClick.add((sender, args) -> {
            System.out.println(blank.getItemName() + " EventMouseClick");
        });
        blank.eventMouseDoubleClick.add((sender, args) -> {
            System.out.println(blank.getItemName() + " EventMouseDoubleClick");
        });
        blank.eventKeyPress.add((sender, args) -> {
            System.out.println(blank.getItemName() + " EventKeyPress");
        });
        blank.eventKeyRelease.add((sender, args) -> {
            System.out.println(blank.getItemName() + " EventKeyRelease");
        });
        // blank.eventMouseHover.add((sender, args) -> {
        //     System.out.println(blank.getItemName() + " EventMouseHover");
        // });
        return blank;
    }

    @Override
    public void initWindow() {

        isBorderHidden = true;
        setSize(400, 400);
        setWindowName("EventTest");
        setWindowTitle("EventTest");

        setMinSize(400, 400);
        isCentered = true;

        setPadding(10, 10, 10, 10);
        // setBackground(new Color(0, 0, 0, 0));
        setBorderRadius(10);
        setAntiAliasingQuality(MSAA.MSAA_8X);
        // isTransparent = true;

        // TitleBar title = new TitleBar("EventTest");
        // title.setShadow(5, 0, 3, new Color(0, 0, 0, 150));
        // addItem(title);

        // Frame cc = new Frame();
        // cc.setMargin(0, title.getHeight() + 10, 0, 0);
        // cc.setBackground(50, 50, 50);
        // addItem(cc);

        BlankItem b1 = getBlankItem("Blank1");
        addItem(b1);
        // BlankItem b2 = getBlankItem("Blank2");
        // BlankItem b3 = getBlankItem("Blank3");

        // cc.addItem(b1);
        // b1.addItem(b2);
        // b2.addItem(b3);

        // b3.setFocus();

        // b3.setPassEvents(false);
        // b2.setPassEvents(false);
        // b2.setPassEvents(true, InputEventType.KEY_PRESS, InputEventType.KEY_RELEASE);

        eventMouseClick.add((sender, args) -> {
            System.out.println(getWindowName() + " EventMouseClick");
        });
        eventMouseDoubleClick.add((sender, args) -> {
            System.out.println(getWindowName() + " EventMouseDoubleClick");
        });
        eventKeyPress.add((sender, args) -> {
            if (args.key == KeyCode.R) {
                // ItemsRefreshManager.printSizeOfShapes();
                // System.out.println(getWindowName() + " EventKeyPress");
            }
        });
        eventKeyRelease.add((sender, args) -> {
            System.out.println(getWindowName() + " EventKeyRelease");
        });
        // getWindow().eventMouseHover.add((sender, args) -> {
        //     System.out.println(getWindow().getItemName() + " EventMouseHover");
        // });
    }
}

File: CSharp/SpaceVIL/SpaceVIL/View/EventTest.cs
using System;
using System.Drawing;
using SpaceVIL;
using SpaceVIL.Core;
using SpaceVIL.Decorations;

namespace View
{
    public class EventTest : ActiveWindow
    {

        private BlankItem GetBlankItem(String name)
        {
            BlankItem blank = new BlankItem();
            blank.SetItemName(name);
            blank.SetStyle(Style.GetFrameStyle());
            blank.SetMargin(50, 50, 50, 50);
            blank.SetBorder(new Border(Color.FromArgb(0, 162, 232), new CornerRadius(10), 3));
            blank.SetBackground(255, 255, 255, 50);
            blank.EventMouseClick += (sender, args) =>
            {
                Console.WriteLine(blank.GetItemName() + " EventMouseClick");
            };
            blank.EventMousePress += (sender, args) =>
            {
                Console.WriteLine(blank.GetItemName() + " EventMousePress");
            };
            // blank.eventMouseDoubleClick += (sender, args) => {
            //     Console.WriteLine(blank.GetItemName() + " EventMouseDoubleClick");
            // });
            blank.EventKeyPress += (sender, args) =>
            {
                Console.WriteLine(blank.GetItemName() + " EventKeyPress");
            };
            blank.EventKeyRelease += (sender, args) =>
            {
                Console.WriteLine(blank.GetItemName() + " EventKeyRelease");
            };
            // blank.eventMouseHover += (sender, args) => {
            //     Console.WriteLine(blank.GetItemName() + " EventMouseHover");
            // });
            return blank;
        }

        public override void InitWindow()
        {

            IsBorderHidden = true;
            SetSize(400, 400);
            SetWindowName("EventTest");
            SetWindowTitle("EventTest");
            SetMinSize(400, 400);
            IsCentered = true;
            SetAntiAliasingQuality(MSAA.MSAA8x);


            TitleBar title = new TitleBar("EventTest");
            title.SetShadow(5, 0, 3, Color.FromArgb(150, 0, 0, 0));
            AddItem(title);

            Frame cc = new Frame();
            cc.SetMargin(0, title.GetHeight() + 10, 0, 0);
            cc.SetBackground(50, 50, 50);
            AddItem(cc);

            BlankItem b1 = GetBlankItem("Blank1");
            BlankItem b2 = GetBlankItem("Blank2");
            BlankItem b3 = GetBlankItem("Blank3");

            cc.AddItem(b1);
            b1.AddItem(b2);
            b2.AddItem(b3);

            b3.SetFocus();

            foreach (var i in b3.GetBlockedEvents())
            {
                Console.WriteLine(i);
            }
        }
    }
}